magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/EC2DiscoveryIT.java

```diff
@@ -0,0 +1,183 @@
+package io.openraven.magpie.plugins.aws.discovery.services;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.openraven.magpie.api.Emitter;
+import io.openraven.magpie.api.MagpieEnvelope;
+import io.openraven.magpie.plugins.aws.discovery.services.base.BaseAWSServiceIT;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.atLeast;
+
+@ExtendWith(MockitoExtension.class)
+public class EC2DiscoveryIT extends BaseAWSServiceIT {
+
+  private static final String CF_EC2_TEMPLATE_PATH = "/template/ec2-template.yml";
+  private EC2Discovery ec2Discovery = new EC2Discovery();
+
+  @Mock
+  private Emitter emitter;
+
+  @Captor
+  private ArgumentCaptor<MagpieEnvelope> envelopeCapture;
+
+  @BeforeAll
+  public static void setup() {
+    updateStackWithResources(CF_EC2_TEMPLATE_PATH);
+  }
+
+  @Test
+  public void testEC2Discovery() {
```

Comment:
whenRunningEc2DiscoveryShouldReturnInstanceEipSecurityGroupVolumeAndSnapshotInformation
It's probably good idea to split this test into 5. Each testing one thing

Reply:
Got your point @PrzemyslawTusinski . We rather have here a complete testing for chosen component from magpie discovery perspective, instead of the unit testing when we could write tests isolated for various corner cases.

Splitting works good but we are going to expose our private methods for component discovery of the various aspects.
For example how we made with IAM - but I dont really like to change private method to protected just for the test availability.

I threat those tests as e2e AWS component discovery based on localstack. It like an IT with WireMocks when you passing from the controller request to database layer and back verifying the response.
In any case I propose to change method naming convention here once we will extend them with additional coverage

Reply:
I didn't meant exposing private methods.
I meant running full discovery in each test case, then during asset phase assert only one resource on each test case